crates/worker/src/partition/snapshots/repository.rs

```diff
@@ -562,33 +564,104 @@ impl SnapshotRepository {
     // own configuration mechanism which doesn't support many of the AWS conventions. This
     // differs quite a lot from the Lambda invoker which uses the AWS SDK, and that would be a
     // very surprising inconsistency for customers. This mechanism allows us to infer the region
     // and securely obtain session credentials without any hard-coded configuration.
     let object_store: Arc<dyn ObjectStore> = if destination.scheme() == "s3" {
-        let builder = if let Ok(profile) = std::env::var("AWS_PROFILE") {
-            // Infer the AWS region from the profile, or use AWS_REGION if set.
-            let region = aws_config::load_defaults(BehaviorVersion::v2024_03_28())
-                .await
-                .region()
-                .context("Unable to determine AWS region")?
-                .to_string();
-            debug!(?profile, ?region, "Using AWS SDK credentials provider");
-            AmazonS3Builder::new()
-                .with_credentials(Arc::new(AwsSdkCredentialsProvider {
-                    credentials_provider: DefaultCredentialsChain::builder().build().await,
-                }))
-                .with_region(region)
+        let builder = match &options.aws_profile {
+            Some(profile) => {
+                debug!(profile, "Using AWS profile for snapshot repository access");
+                let sdk_config = aws_config::defaults(BehaviorVersion::v2024_03_28())
+                    .profile_name(profile)
+                    .load()
+                    .await;
+
+                let region = DefaultRegionChain::builder()
+                    .profile_name(profile)
+                    .build()
+                    .region()
+                    .await
+                    .context("Unable to determine region from profile")?;
+
+                debug!(?region, ?profile, "Using AWS SDK credentials provider");
+                let creds = DefaultCredentialsChain::builder()
+                    .profile_name(profile)
+                    .region(region.clone())
+                    .build()
+                    .await;
+
+                let builder = AmazonS3Builder::new()
+                    .with_credentials(Arc::new(AwsSdkCredentialsProvider {
+                        credentials_provider: creds,
+                    }))
+                    .with_region(region.to_string());
+
+                if let Some(endpoint_url) = sdk_config.endpoint_url() {
+                    debug!(endpoint_url, "Using custom AWS endpoint from profile");
+                    builder.with_endpoint(endpoint_url)
+                } else {
+                    builder
+                }
+            }
+            None => {
+                let sdk_config = aws_config::load_defaults(BehaviorVersion::v2024_03_28()).await;
+
+                let region = sdk_config
+                    .region()
+                    .context("Unable to determine AWS region")?
+                    .to_string();
```

Comment:
What's great about this approach is that if the customer doesn't set an explicit `[worker.snapshots] aws_profile = <profile>` to use, but have a default config profile - or have exported a  `AWS_PROFILE` - we'll just pick those up, same as any other AWS SDK-based tool.

Reply:
What if I have configured a region via `aws_config`? Or is it invalid to configure a profile w/o a region?